rfcs/text/0352-graph-type-system.md

```diff
@@ -0,0 +1,2659 @@
+
+### Data Types
+
+A **Data Type** describes a space of possible valid _values_. For instance, the _string_ value `"foo"` might be an instance of a `Text` Data Type, or a _float_ value `0.6` might be an instance of a `Number` Data Type ("instance of" means that the value satisfies the constraints expressed on the Data Type, or alternatively you can view it as saying the value is within the value space of the Data Type).
+
+> 💬 The underlying value's representation will depend on the implementation environment, different languages and technologies may store these differently.
+
+A Data Type is composed of the following:
+
+- a **required** `$id`, which is a globally unique identifier, where the Data Type's definition can be accessed (in most cases this will be a URL)
+- a **required** `name`, which should generally be a non-pluralized short description of the Data Type (for example: "Number" not "Numbers")
+- **optionally** a `description` to further explain the semantic meaning of the Data Type
+- a **required** definition of its possible value
+
+#### Primitive Top-Level Data Types
+
+We pre-define a set of primitive top-level Data Types.
+These can be considered the least restrictive value spaces:
+
+- `Text`
+- `Number`
+- `Boolean`
+- `Null`
+- `Object`
+- `Empty List` (see [here](#including-an-empty-list-data-type) for why this is a top-level Data Type)
+
+#### Creating new Data-Types
+
+Due to the size of this RFC, creating new data-types has been deemed out-of-scope and a [follow-up RFC](https://github.com/blockprotocol/blockprotocol/pull/355) is in the works to spec out how this could work. Please consult it for plans on introducing a suite of Data Types including ones like `Date`, `Positive Number`, etc.
+
+### Property Types
+
+A **Property Type** is a description of a named piece of data, including its possible values. A Property Type is composed of the following:
+
+- a **required** `$id`, which is a globally unique identifier, where the Property Type's definition can be accessed (in most cases this will be a URL)
+- a **required** `name`, which should generally be a non-pluralized description of the property (for example: "Address" not "Addresses")
+- **optionally** a `description` to further explain the semantic meaning of the Property Type
+- a **required** definition of its possible values
+
+**Defining the possible values of a property**
+
+#### Data Types
+
+A Property Type can have plain data as its value, which it can define by using a Data Type.
+
+**Example 1**
+
+The `Favorite Quote` Property Type could define its value as being an instance of the `Text` Data Type.
+
+- Sample data when used in (simplified view of) an Entity
+
+  ```json
+  {
+    "favoriteQuote": "He who controls the spice controls the universe"
```

Comment:
I was worried that the simplification would be confusing, thank you for confirming.

Could we fix this by adding a big ol' disclaimer saying that the examples in this section are not representative and use the name and blah blah blah?

I think there's still value in understanding how these types describe data, and how it isn't _too_ different from the current way of doing things beyond people needing to rewrite keys. I'm not sure how I feel about not giving those high-level explanatory examples in this section and forcing people to be faced with the Reference-Level which really is a lot less approachable.

If a disclaimer wouldn't really work perhaps there's something else we can do, if you think the data examples don't actually give much value I'm content with dropping them.

Reply:
I think using non-concrete URIs would help.

> Could we fix this by adding a big ol' disclaimer saying that the examples in this section are not representative and use the name and blah blah blah?

I don't think this will help because the examples do not use the name. You could say 'the examples use a camel case version of the name', but then it becomes even more explicitly 'Here are some examples of things which we aren't proposing'

Reply:
Added non-concrete URIs in https://github.com/blockprotocol/blockprotocol/pull/352/commits/30d59ba15a0e4c415fa25f065be90aa306380004